opm/parser/eclipse/EclipseState/EclipseState.hpp

```diff
@@ -153,6 +153,12 @@ namespace Opm {
         const std::map<int, VFPInjTable>& getVFPInjTables() const;
         size_t getNumPhases() const;
 
+        // The saturation function family.
+        // If SWOF and SGOF are specified in the deck it return 1
+        // If SWFN, SGFN and SOF3 are specified in the deck it return 2
+        // If keywords are missing or mixed, an error is given.
+        const size_t getSaturationFunctionFamily() const;
```

Comment:
> Just curious: why is it needed in opm-material? Is this not abstracted away well enough by the EclipseState?

that's because the columns of the tables must be initialized differently for every family. you can argue that this should be done into opm-parser, but if this is supposed to simplify matters in consumer modules, there must be a canonical representation for all families and I'm not sure if this is (easily) possible.